pandas/core/groupby/ops.py

```diff
@@ -157,7 +157,7 @@ def apply(self, f, data, axis: int = 0):
 
         elif (
             com.get_callable_name(f) not in base.plotting_methods
-            and hasattr(splitter, "fast_apply")
+            and splitter.fast_apply is not None
```

Comment:
I can see why mypy complains about this but the new way of doing this might be fragile as well. Would making this an abstract property in the base class be difficult?

Reply:
I dont think mypy is complaining about this, but ill change this check to be clearer on that front